engine/executor/config.go

```diff
@@ -216,3 +216,10 @@ func (c *Config) configFromFile(path string) error {
 	}
 	return nil
 }
+
+func getDefaultLocalStorageDir(executorName string) string {
+	// Use hex encoding in case there are special characters in the
+	// executor name.
+	encodedExecutorName := hex.EncodeToString([]byte(executorName))
+	return filepath.Join(defaultLocalStorageDirPrefix, encodedExecutorName)
```

Comment:
Do we need to check existence of the path, or use ioutil.TempDir to make sure always use a new directory?